core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -226,7 +228,11 @@ public InterceptorChain buildInterceptorChain() {
          interceptorChain.appendInterceptor(createInterceptor(new GroupingInterceptor(), GroupingInterceptor.class), false);
       }
 
-      if (needsVersionAwareComponents && configuration.clustering().cacheMode().isClustered()) {
+      // The expiration interceptor must come before the entry wrapping interceptor as it can stop data container
+      // access from expiring things.
```

Comment:
Nothing it would just delay the expiration from occuring until after the write was unregistered.